crates/bevy_core_widgets/src/core_slider.rs

```diff
@@ -0,0 +1,404 @@
+) {
+    if let Ok((_slider, mut drag)) = q_slider.get_mut(trigger.target().unwrap()) {
+        trigger.propagate(false);
+        if drag.dragging {
+            drag.dragging = false;
+        }
+    }
+}
+
+fn slider_on_key_input(
+    mut trigger: Trigger<FocusedInput<KeyboardInput>>,
+    q_slider: Query<(
+        &CoreSlider,
+        &SliderValue,
+        &SliderRange,
+        &SliderStep,
+        Has<InteractionDisabled>,
+    )>,
+    mut commands: Commands,
+) {
+    if let Ok((slider, value, range, step, disabled)) = q_slider.get(trigger.target().unwrap()) {
+        let event = &trigger.event().input;
+        if !disabled && event.state == ButtonState::Pressed {
+            let new_value = match event.key_code {
+                KeyCode::ArrowLeft => range.clamp(value.0 - step.0),
+                KeyCode::ArrowRight => range.clamp(value.0 + step.0),
+                KeyCode::Home => range.start(),
+                KeyCode::End => range.end(),
+                _ => {
+                    return;
+                }
+            };
+            trigger.propagate(false);
+            if let Some(on_change) = slider.on_change {
+                commands.run_system_with(on_change, new_value);
+            }
+        }
+    }
+}
+
+pub(crate) fn slider_on_insert(trigger: Trigger<OnInsert, CoreSlider>, mut world: DeferredWorld) {
+    let mut entity = world.entity_mut(trigger.target().unwrap());
+    if let Some(mut accessibility) = entity.get_mut::<AccessibilityNode>() {
+        accessibility.set_orientation(Orientation::Horizontal);
+    }
+}
+
+pub(crate) fn slider_on_insert_value(
+    trigger: Trigger<OnInsert, SliderValue>,
+    mut world: DeferredWorld,
+) {
+    let mut entity = world.entity_mut(trigger.target().unwrap());
+    let value = entity.get::<SliderValue>().unwrap().0;
+    if let Some(mut accessibility) = entity.get_mut::<AccessibilityNode>() {
+        accessibility.set_numeric_value(value.into());
+    }
+}
+
+pub(crate) fn slider_on_insert_range(
+    trigger: Trigger<OnInsert, SliderRange>,
+    mut world: DeferredWorld,
+) {
+    let mut entity = world.entity_mut(trigger.target().unwrap());
+    let range = entity.get::<SliderRange>().unwrap().0.clone();
+    if let Some(mut accessibility) = entity.get_mut::<AccessibilityNode>() {
+        accessibility.set_min_numeric_value((*range.start()).into());
+        accessibility.set_max_numeric_value((*range.end()).into());
+    }
+}
+
+pub(crate) fn slider_on_insert_step(
+    trigger: Trigger<OnInsert, SliderStep>,
+    mut world: DeferredWorld,
+) {
+    let mut entity = world.entity_mut(trigger.target().unwrap());
+    let step = entity.get::<SliderStep>().unwrap().0;
+    if let Some(mut accessibility) = entity.get_mut::<AccessibilityNode>() {
+        accessibility.set_numeric_value_step(step.into());
+    }
+}
+
+/// Event which can be triggered on a slider to modify the value (using the `on_change` callback).
```

Comment:
This would really benefit from a doc test.

Reply:
Done, but TBH it's not much of a test. A more comprehensive test would require more boilerplate, since we'd need to do a sync point to see if the command actually worked.

Reply:
Yeah, explanation is the key thing here; we can do real tests elsewhere if needed.